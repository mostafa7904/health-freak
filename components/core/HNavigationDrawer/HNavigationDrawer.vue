<template>
  <nav :class="computedClasses">
    <slot name="prepend" class="h-navigation-drawer__prepend"> </slot>
    <slot class="h-navigation-drawer__content"></slot>
    <slot name="append" class="h-navigation-drawer__append"></slot>
  </nav>
</template>

<script lang="ts">
import { PropType, defineComponent } from 'vue'

export default defineComponent({
  name: 'HealthFreakNavigationDrawer',
  props: {
    value: null as unknown as PropType<any>
  },
  computed: {
    computedClasses(): any {
      return {
        'h-navigation-drawer': true,
        'h-navigation-drawer--open': this.value,
        'h-navigation-drawer--close': !this.value
      }
    }
  }
})
</script>

<style lang="scss">
.h-navigation-drawer {
  position: fixed;
  left: 0;
  top: 0;
  display: flex;
  flex-direction: column;
  background: white;
  width: 100%;
  height: 100vh;
  overflow: auto;
  z-index: 999;
  transition-duration: 0.2s;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-property: transform, visibility, width;
  &__content {
    height: 100%;
    overflow-y: auto;
    overflow-x: hidden;
  }
  &--open {
    transform: translateX(0);
    html,
    body {
      overflow-y: hidden;
    }
  }
}
html[lang='fa_IR'] {
  .h-navigation-drawer--close {
    transform: translateX(100%);
  }
}
html[lang='en_US'] {
  .h-navigation-drawer--close {
    transform: translateX(-100%);
  }
}
</style>
