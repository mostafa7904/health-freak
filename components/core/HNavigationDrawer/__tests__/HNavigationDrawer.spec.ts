// Libraries
import Vue from 'vue'

// Utilities
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

// Components
import HNavigationDrawer from '../HNavigationDrawer.vue'

describe('HNavigationDrawer.vue', () => {
  let mountFunction: (options?: object) => Wrapper<Vue>

  let localVue: typeof Vue

  beforeEach(() => {
    localVue = createLocalVue()

    mountFunction = (options = {}) => {
      return mount(HNavigationDrawer, {
        localVue,
        ...options
      })
    }
  })

  it('should render navigation drawer closed if no value is passed', () => {
    const html = mountFunction().html()
    expect(html).toMatchSnapshot()
  })

  it('should render navigation drawer open if value passed is true', () => {
    const html = mountFunction({
      propsData: {
        value: true
      }
    }).html()
    expect(html).toMatchSnapshot()
  })

  it('should change the state if the value is changed', async () => {
    const wrapper = mountFunction({
      propsData: {
        value: true
      }
    })
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({ value: false })
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })
})
