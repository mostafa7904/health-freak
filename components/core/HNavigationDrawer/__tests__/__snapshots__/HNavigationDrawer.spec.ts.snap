// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HNavigationDrawer.vue should change the state if the value is changed 1`] = `"<nav class=\\"h-navigation-drawer h-navigation-drawer--open\\"> </nav>"`;

exports[`HNavigationDrawer.vue should change the state if the value is changed 2`] = `"<nav class=\\"h-navigation-drawer h-navigation-drawer--close\\"> </nav>"`;

exports[`HNavigationDrawer.vue should render navigation drawer closed if no value is passed 1`] = `"<nav class=\\"h-navigation-drawer h-navigation-drawer--close\\"> </nav>"`;

exports[`HNavigationDrawer.vue should render navigation drawer open if value passed is true 1`] = `"<nav class=\\"h-navigation-drawer h-navigation-drawer--open\\"> </nav>"`;
